ledger/db/src/tx_out_store.rs

```diff
@@ -1222,17 +1241,14 @@ pub mod tx_out_store_tests {
 
             */
 
-            assert_eq!(7, proof.elements.len());
+            assert_eq!(4, proof.elements.len());
```

Comment:
This might be a good place to test that the elements are in the expected order -- I don't think I've seen that tested specifically in this PR. Something like

```
let expected_ranges = vec![Range::new(3, 3).unwrap(), Range::new(2, 2).unwrap(), Range::new(0, 1).unwrap(), Range::new(4, 7).unwrap()];
assert_eq!(ranges, expected_ranges);
```

Bonus points for updating the ASCII art right above this to explain or illustrate that the hashes of internal nodes are not included because they are implied.